lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -1167,35 +1210,67 @@ struct AIEObjectFifoStatefulTransformPass
+        DimTupleArrayAttr emptyDims =
+            DimTupleArrayAttr::get(builder.getContext(), {});
+        DimTupleArrayAttr singletonFromStreamDims = DimTupleArrayAttr::get(
+            builder.getContext(), {consumerDims[consumerIndex]});
+        DimTupleArrayArrayAttr fromStreamDims = DimTupleArrayArrayAttr::get(
+            builder.getContext(), singletonFromStreamDims);
+        ObjectFifoCreateOp consumerFifo = createObjectFifo(
+            builder, datatype, consumerFifoName, consumerTile, consumerTile,
+            consumerObjFifoSize, emptyDims, fromStreamDims);
         replaceSplitFifo(createOp, consumerFifo, consumerTileOp);
 
-        // identify external buffers that were registered to
-        // the consumer objectFifo
+        // identify external buffers that were registered to the consumer fifo
         if (consumerTile.getDefiningOp<TileOp>().isShimTile())
           detectExternalBuffers(device, createOp, consumerFifo, consumerTile);
 
-        // record that this objectFifo was split
+        // record that this objectFifo was split; it will require DMA config
         splitConsumerFifos.push_back(consumerFifo);
 
         // update the linkOp if the split objFifo was originally its start point
         auto linkOp = getOptionalLinkOp(createOp);
-        if (linkOp)
-          for (auto fifoIn : linkOp->getInputObjectFifos())
-            if (fifoIn.name() == createOp.name())
+        if (linkOp) {
+          for (auto fifoIn : linkOp->getInputObjectFifos()) {
+            if (fifoIn.name() == createOp.name()) {
               if (consumerTile == *(linkOp->getOptionalSharedTile())) {
                 auto res = mlir::SymbolTable::replaceAllSymbolUses(
                     createOp.name(), consumerFifo.name(),
                     linkOp->getOperation());
                 if (res.failed())
                   llvm_unreachable("unreachable");
               }
+            }
+          }
+        }
+
+        consumerIndex++;
+      }
+
+      if (splitConsumerFifos.size() > 0) {
+        splitFifos.push_back({createOp, splitConsumerFifos});
+      }
+    }
+
+    //===------------------------------------------------------------------===//
+    // - Create objectFifo buffers and locks.
+    // - Populate a list of tiles containing objectFifos for later processing of
+    //   the acquires/releases (uses of the FIFO).
+    //===------------------------------------------------------------------===//
+    for (auto createOp : device.getOps<ObjectFifoCreateOp>()) {
+      objectFifoTiles.insert(createOp.getProducerTileOp());
+      bool shared = false;
+      int share_direction = 0;
+      for (auto consumerTile : createOp.getConsumerTiles()) {
+        TileOp consumerTileOp = dyn_cast<TileOp>(consumerTile.getDefiningOp());
+        objectFifoTiles.insert(consumerTileOp);
+        if (!requiresDMAs(createOp, consumerTileOp, share_direction)) {
+          shared = true;
```

Comment:
This mirrors what your code did previously (plus my changes): If there is more than one consumer, `requiresDMAs` returns `true`, we do not enter the block, so `shared=false`. One way to read `!requiresDMAs` (note the `!`) would be `canUseSharedMemory`, which is not the case if we are broadcasting.

Reply:
Yes, looking at this code again, I believe that our objectFifo implementation cannot correctly handle an objectFifo with one memory-adjacent and one "far-away" consumer (i.e. one has shared-memory and one requires DMA). Reason for this being this `break` (and also setting `shared=True` if only one of the consumers has shared memory -- it should be if _all_ have shared memory). Maybe this can be addressed in another PR as the issue would have existed before my changes and is orthogonal to them.

Reply:
I agree that a 'continue' would be more appropriate and also solve the "ensuring all consumer tiles are taken into account". 

But it's true that I tested this (broadcast where every consumer has a loop to unroll) and it worked with the 'break' too....

Reply:
If you find any errors that are orthogonal to this, you can create issues and assign them to me! That's a lot of help! :)